src/queue.rs

```diff
@@ -262,11 +265,14 @@ impl Queue {
             let mut current = self.current_track.write().unwrap();
             current.replace(index);
             self.spotify.update_track();
+
+            #[cfg(feature = "notify")]
             if self.cfg.values().notify.unwrap_or(false) {
-                #[cfg(feature = "notify")]
-                if let Err(e) = Notification::new().summary(&track.to_string()).show() {
-                    error!("error showing notification: {:?}", e);
-                }
+                std::thread::spawn({
+                    let track_name = track.to_string().clone();
+                    let cover_url = track.cover_url().unwrap().clone();
```

Comment:
Is it guaranteed that Spotify will always provide a `cover_url`? If not, this `unwrap()` may panic.